### ISSUE TITLE
UI fallback

### ISSUE THREAD

**Issue by enumag** [07th-mod/higurashi-assembly#28]
Sometimes the UI mod gets broken after a new release from MG. It would be nice if we could detect this based on the unify version of the game and the UI file - the game does have both of these numbers because they can be found in the log file after such error.

When we detect different unity version we should instead load the vanilla UI which was renamed to `.backup` when installing the mod. Also we should print some message to the screen or into a popup to let the user know what happened and that they should report the issue to us on GitHub or Discord.

**Comment by enumag** [comment]
@idealpersona These errors happen more commonly after the recent updates so this is now a high priority issue which I'd really like to have solved for the next release (which should be finished this month). Could you look into it?

cc @ItaloKnox

**Comment by ItaloKnox** [comment]
I talked to drojf yesterday about this and for now, we're polishing the UI backups a bit and adding an option to restore the original UI and have it sent to us.

At the same time, I believe we should work on changing the version to the UI suffix as we talked a few days ago. The installer should then parse the Unity version from the user and autocomplete the link to check if it's valid. If it's valid, download the UI. If it's not, skip it. I believe that should be a decent fix for now.

**Comment by ItaloKnox** [comment]
Actually, just noticed we're not in the installer repo and this was the dll solution we were talking about a long time ago. If that can be done, I would prefer it over the tedious method of letting the installer always handle it.

Also, let me tag @drojf, then.

**Comment by drojf** [comment]
uhhh....I'll just await the .dll people to respond then. Here is a link to the corresponding issue raised on the installer so we can keep track of things: https://github.com/07th-mod/python-patcher/issues/38

I believe @tellowkrinkle  already detects the unity version and only installs the UI based on that, could be wrong (see below commit). The other day I tried deleting all UI overrides while I was testing something and it didn't let me install, so I'm not sure if it lets you install if there is no compatible UI pack for your Unity version.

See this commit: https://github.com/07th-mod/python-patcher/commit/57eff70ac03020cf7034fd353a62d95c7fd84e01

**Comment by tellowkrinkle** [comment]
Yeah with that change the installer will no longer allow the install to continue if there's a Unity version mismatch

**Comment by enumag** [comment]
Hmm I think we need to push my new UI packs. Here is how it's supposed to work in my opinion:

- The new packs have a unity version in the file name, for example `Onikakushi-UI_5.2.2f1_win.7z`. I have these ready on my PC.
- I written a smart PHP script which will allow the installer to use urls like this: `http://07th-mod.com/download_ui.php?chapter=onikakushi&system=win&unity=5.2.2f1`
- The script looks if the UI with correct version exists and redirects to that file.
- If it does not exist it throws 404.
- In that case the installer should notify the user and possibly have an option to upload his vanilla UI files for us to process.
